torch/utils/checkpoint.py

```diff
@@ -187,18 +187,26 @@ def add_device_ids(arg):
     return fwd_device_ids, fwd_device_states
 
 
-def set_device_states(devices, states) -> None:
-    device_module = _get_device_module(_infer_device_type(*states))
+def set_device_states(devices, states, *, device_type='cuda') -> None:
+    """Sets random number generator states for the specified devices.
+
+    Args:
+        devices: Device ids to set states for.
+        states: States to set.
+        device_type: ``device_type`` of the devices to set states for. This is
+            required in case of non-cuda devices. Default is ``cuda``.
+    """
+    device_module = _get_device_module(device_type)
```

Comment:
Sorry I missed this earlier, but to preserve existing behavior, we should have the arg default to None, and take the value of  `DefaultDeviceType.get_device_type()` instead.

Reply:
> take the value of DefaultDeviceType.get_device_type() instead.

Done.